From 000814078971df6334abd62157756e7eb588f8ca Mon Sep 17 00:00:00 2001
From: Harish Raja Selvan <harish.rajaselvan@multicorewareinc.com>
Date: Wed, 20 May 2026 01:06:47 -0700
Subject: [PATCH] ARM64EC libc++ changes

---
 include/__threading_support | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/__threading_support b/include/__threading_support
index 8f1efb785..1deda550b 100644
--- a/include/__threading_support
+++ b/include/__threading_support
@@ -125,7 +125,7 @@ typedef void* __libcpp_mutex_t;
 
 #if defined(_M_IX86) || defined(__i386__) || defined(_M_ARM) || defined(__arm__)
 typedef void* __libcpp_recursive_mutex_t[6];
-#elif defined(_M_AMD64) || defined(__x86_64__) || defined(_M_ARM64) || defined(__aarch64__)
+#elif defined(_M_AMD64) || defined(__x86_64__) || defined(_M_ARM64) || defined(__aarch64__) || defined(_M_ARM64EC)
 typedef void* __libcpp_recursive_mutex_t[5];
 #else
 # error Unsupported architecture
-- 
2.53.0.windows.1

